---
title: "RNCBIGene: NCBI Gene annotation in parquet"
shorttitle: "NCBI Gene for all organisms"
author: "Vincent J. Carey, stvjc at channing.harvard.edu"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{RNCBIGene: NCBI Gene annotation in parquet}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    highlight: pygments
    number_sections: yes
    theme: united
    toc: yes
---

# Introduction

Bioconductor's annotation system has functioned for well
over a decade as a pivotal resource for analysis and
tool development.  See the [AnnotationDbi](https://bioconductor.org/packages/AnnotationDbi)
documentation for details.  AnnotationDbi and packages based
on it use SQLite to manage
organism-specific vocabularies and to resolve queries about
mappings between identifier sets.  For example,
learn the Gene Ontology annotations for gene ORMDL3
in _Homo sapiens_:

```{r lkorg1, message=FALSE}
library(org.Hs.eg.db)
go_orm = AnnotationDbi::select(org.Hs.eg.db, key="ORMDL3", keytype="SYMBOL",
    columns=c("GO", "ENTREZID"))
head(go_orm)
```

With RNCBIGene, all NCBI's GO annotations for all organisms are in a
single resource.  The `geneFromCache` package will retrieve
an Apache Parquet representation of the gene2go.gz text
resource distributed at NCBI's FTP site.  For this
query, we use the knowledge the ORMDL3 has Gene ID 94103.

```{r lknc1,message=FALSE}
library(RNCBIGene)
library(dplyr)
go_ds = arrow::open_dataset(geneFromCache("gene2go.parquet"))
lkor = go_ds |> filter(`#tax_id`==9606, GeneID==94103) |> collect() |> as.data.frame()
DT::datatable(lkor)
```

It is not the purpose of this package to provide pin-compatible
replacements for the AnnotationDbi packages.  That would
be a huge effort, and material support for such an
effort does not exist at this
time.  This package is created to explore the opportunities
for simplification and performance enhancement 
arising from the adoption of Parquet and Arrow for annotation
representation and interrogation.

# Scope

There are seven parquet files that represent
the NCBI Gene content available on February 22 2025.
```{r lksev}
RNCBIGene:::available_gene_parquet
```

Once we have cached them using
`geneFromCache`, we can work with all of them "simultaneously".

Because the retrieval process is laborious, the following chunk
is not evaluated.  Were it to be evaluated, it would be
apparent that over 720 million records of annotation
are available for interrogation.

```{r lksim, eval=FALSE}
paths = lapply(RNCBIGene:::available_gene_parquet, geneFromCache)
pop = lapply(paths, arrow::open_dataset)
names(pop) = RNCBIGene:::available_gene_parquet
sapply(pop, nrow)
```

For illustration, the content corresponding to four human
genes on chromosome 17 has been saved in the package.

```{r lklit}
litp = dir(system.file("litparquet", package="RNCBIGene"), full.names=TRUE)
litpop = lapply(litp,
    arrow::open_dataset)
ns = basename(litp)
names(litpop) = ns
sapply(litpop, nrow)
```

Example: find the UniProt identifiers associated
with gene ORMDL3 in humans.

```{r lkuni}
pro = litpop[["lit.gene2accession.parquet"]] |> 
   filter(`#tax_id` == 9606, Symbol == "ORMDL3") |> 
   select(protein_accession.version) |> collect() |> pull() |> setdiff("-")
litpop[["lit.gene_refseq_uniprotkb_collab.parquet"]] |> filter(NCBI_tax_id == 9606) |> 
   filter(`#NCBI_protein_accession` %in% pro)  |> collect()
```

# Applications

## Tidyverse-oriented annotation mapping

Typically one would use org.Hs.eg.db to remap identifiers
from a SummarizedExperiment.  We want to be able to do
this for any identifier used in NCBI annotation, and for
any organism.  Let's start with a dataset for human
airway transcriptomics.  The features are annotated
with Ensembl gene identifiers, and we will add
map locations and MIM codes using `mapIdsNG`, a variant
of AnnotationDbi's mapIds functionality for NCBI Gene.

```{r dotidy}
data(airway, package="airway")
requireNamespace("tidySummarizedExperiment")
tse = as(airway, "tidySummarizedExperiment")
tse = tse |> dplyr::mutate(map_location=mapIdsNG(keys=.feature, keytype="Ensembl", column="map_location"))
tse = tse |> dplyr::mutate(MIM=mapIdsNG(keys=.feature, keytype="Ensembl", column="MIM"))
head(table(SummarizedExperiment::rowData(tse)$map_location))
```

Note that the identifiers that are conveniently available for
`mapIdsNG` are currently limited to those in `gene_info`:
```{r lkids}
colnames(litpop[["lit.gene_info.parquet"]])
```



# Session information

```{r lksess}
sessionInfo()
```
