<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>High-performance data representations of gene annotations from NCBI • RNCBIGene</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="High-performance data representations of gene annotations from NCBI">
<meta name="description" content="This package is based on transforming key textual resources from NCBI Gene to parquet. Resources from May 2025 were made available in an NSF Open Storage Network bucket. This package illustrates the use of standard and tidyverse-oriented approaches to querying and using annotation on genes for all organisms covered by NCBI Gene.">
<meta property="og:description" content="This package is based on transforming key textual resources from NCBI Gene to parquet. Resources from May 2025 were made available in an NSF Open Storage Network bucket. This package illustrates the use of standard and tidyverse-oriented approaches to querying and using annotation on genes for all organisms covered by NCBI Gene.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">RNCBIGene</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/RNCBIGene.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/vjcitn/RNCBIGene/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="rncbigene">RNCBIGene<a class="anchor" aria-label="anchor" href="#rncbigene"></a>
</h1></div>
<p>Performant and standard representations of gene annotation for all organisms cataloged by NCBI</p>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p><code>devtools::install_github("vjcitn/RNCBIGene")</code></p>
</div>
<div class="section level1">
<h1 id="purposes">Purposes<a class="anchor" aria-label="anchor" href="#purposes"></a>
</h1>
<div class="section level2">
<h2 id="simplified-unified-annotation-for-all-organisms-addressed-by-ncbi">Simplified, unified annotation for all organisms addressed by NCBI<a class="anchor" aria-label="anchor" href="#simplified-unified-annotation-for-all-organisms-addressed-by-ncbi"></a>
</h2>
<p>The <code>org.*.*.db</code> packages are powerful and reliable but have a complex stack of schemata and scripts for generating organism-specific packages.</p>
<p>This package works the basis of a one-line transformation to parquet of compressed text from NCBI. The parquet files were placed in an NSF Open Storage Network bucket.</p>
<p>The <code>geneFromCache</code> function retrieves and caches the parquet files.</p>
<pre><code>&gt; example(geneFromCache)

gnFrmC&gt; gi = geneFromCache("gene_info.parquet")

gnFrmC&gt; arrow::open_dataset(gi) |&gt; dplyr::filter(`#tax_id`==9606) |&gt; head() |&gt; dplyr::collect()
# A tibble: 6 × 16
  `#tax_id` GeneID Symbol LocusTag Synonyms      dbXrefs chromosome map_location
      &lt;int&gt;  &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;    &lt;chr&gt;         &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;
1      9606      1 A1BG   -        A1B|ABG|GAB|… MIM:13… 19         19q13.43
2      9606      2 A2M    -        A2MD|CPAMD5|… MIM:10… 12         12p13.31
3      9606      3 A2MP1  -        A2MP          HGNC:H… 12         12p13.31
4      9606      9 NAT1   -        AAC1|MNAT|NA… MIM:10… 8          8p22
5      9606     10 NAT2   -        AAC2|NAT-2|P… MIM:61… 8          8p22
6      9606     11 NATP   -        AACP|NATP1    HGNC:H… 8          8p22
# ℹ 8 more variables: description &lt;chr&gt;, type_of_gene &lt;chr&gt;,
#   Symbol_from_nomenclature_authority &lt;chr&gt;,
#   Full_name_from_nomenclature_authority &lt;chr&gt;, Nomenclature_status &lt;chr&gt;,
#   Other_designations &lt;chr&gt;, Modification_date &lt;int&gt;, Feature_type &lt;chr&gt;</code></pre>
<p>Caching the resources involves transfers of 6GB of parquet; if this is too laborious, remote queries can be executed for specific needs:</p>
<pre><code>&gt; x = remote_gene_query(gres="gene_info", qual='where "#tax_id" = 9606 limit 10')
&gt; x
# A tibble: 10 × 16
   `#tax_id` GeneID Symbol   LocusTag Synonyms   dbXrefs chromosome map_location
       &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;
 1      9606      1 A1BG     -        A1B|ABG|G… MIM:13… 19         19q13.43
 2      9606      2 A2M      -        A2MD|CPAM… MIM:10… 12         12p13.31
 3      9606      3 A2MP1    -        A2MP       HGNC:H… 12         12p13.31
 4      9606      9 NAT1     -        AAC1|MNAT… MIM:10… 8          8p22
 5      9606     10 NAT2     -        AAC2|NAT-… MIM:61… 8          8p22
 6      9606     11 NATP     -        AACP|NATP1 HGNC:H… 8          8p22
 7      9606     12 SERPINA3 -        AACT|ACT|… MIM:10… 14         14q32.13
 8      9606     13 AADAC    -        CES5A1|DAC MIM:60… 3          3q25.1
 9      9606     14 AAMP     -        -          MIM:60… 2          2q35
10      9606     15 AANAT    -        DSPS|SNAT  MIM:60… 17         17q25.1
# ℹ 8 more variables: description &lt;chr&gt;, type_of_gene &lt;chr&gt;,
#   Symbol_from_nomenclature_authority &lt;chr&gt;,
#   Full_name_from_nomenclature_authority &lt;chr&gt;, Nomenclature_status &lt;chr&gt;,
#   Other_designations &lt;chr&gt;, Modification_date &lt;dbl&gt;, Feature_type &lt;chr&gt;</code></pre>
</div>
<div class="section level2">
<h2 id="annotation-in-a-tidyverse-style">Annotation in a tidyverse style<a class="anchor" aria-label="anchor" href="#annotation-in-a-tidyverse-style"></a>
</h2>
<p>By retaining the “flat file” model of the original all-organism annotation content at NCBI, we may more straightforwardly have access to annotation mappings in tidyverse-style programming. As an example, given TCGA expression data annotated with gene symbols, we can add the Gene (Entrez) Ids as follows.</p>
<pre><code>&gt; suppressMessages({
+ gbt = curatedTCGAData(diseaseCode="GBM",
+   assays="RNASeq2GeneNorm", version="2.0.1", dry.run=FALSE,
+   verbose=FALSE)
+ })
&gt; gbtr = experiments(gbt)[[1]]
&gt; library(tidyomics)
&gt; library(RNCBIGene)
&gt; gbtr[1:10,]
# A SummarizedExperiment-tibble abstraction: 1,660 × 2
# Features=10 | Samples=166 | Assays=
   .feature .sample
   &lt;chr&gt;    &lt;chr&gt;
 1 A1BG     TCGA-02-0047-01A-01R-1849-01
 2 A1CF     TCGA-02-0047-01A-01R-1849-01
 3 A2BP1    TCGA-02-0047-01A-01R-1849-01
 4 A2LD1    TCGA-02-0047-01A-01R-1849-01
 5 A2ML1    TCGA-02-0047-01A-01R-1849-01
 6 A2M      TCGA-02-0047-01A-01R-1849-01
 7 A4GALT   TCGA-02-0047-01A-01R-1849-01
 8 A4GNT    TCGA-02-0047-01A-01R-1849-01
 9 AAA1     TCGA-02-0047-01A-01R-1849-01
10 AAAS     TCGA-02-0047-01A-01R-1849-01
# ℹ 190 more rows
# ℹ Use `print(n = ...)` to see more rows

&gt; gbtr2 = gbtr[1:1000,] |&gt;
+    mutate(entrez = mapIdsNG(keys=.feature)$GeneID)
&gt; gbtr2[1:10,]
# A SummarizedExperiment-tibble abstraction: 1,660 × 3
# Features=10 | Samples=166 | Assays=
   .feature .sample                         entrez
   &lt;chr&gt;    &lt;chr&gt;                            &lt;dbl&gt;
 1 A1BG     TCGA-02-0047-01A-01R-1849-01         1
 2 A1CF     TCGA-02-0047-01A-01R-1849-01     29974
 3 A2BP1    TCGA-02-0047-01A-01R-1849-01        NA
 4 A2LD1    TCGA-02-0047-01A-01R-1849-01        NA
 5 A2ML1    TCGA-02-0047-01A-01R-1849-01    144568
 6 A2M      TCGA-02-0047-01A-01R-1849-01         2
 7 A4GALT   TCGA-02-0047-01A-01R-1849-01     53947
 8 A4GNT    TCGA-02-0047-01A-01R-1849-01     51146
 9 AAA1     TCGA-02-0047-01A-01R-1849-01 100329167
10 AAAS     TCGA-02-0047-01A-01R-1849-01      8086
# ℹ 190 more rows
# ℹ Use `print(n = ...)` to see more rows</code></pre>
<p>The restriction to the first 1000 features in the example above arises because an unrestricted attempt fails for a reason that is currently obscure.</p>
</div>
</div>
<div class="section level1">
<h1 id="available-resources">Available resources<a class="anchor" aria-label="anchor" href="#available-resources"></a>
</h1>
<p>Contents of <code>https://mghp.osn.xsede.org/bir190004-bucket01/BiocParquetNCBI</code>, as retrieved on 22 Feb 2025 and then transformed to parquet (see inst/scripts for demonstration code):</p>
<pre><code>3002338275 2025-05-17 09:28:29.618223689 gene2accession.parquet
680657744 2025-05-15 10:31:33.209868982 gene2go.parquet
 89080561 2025-05-15 10:37:45.825464461 gene2pubmed.parquet
1467468877 2025-05-17 09:28:38.701230380 gene2refseq.parquet
965232147 2025-05-15 10:33:17.845775772 gene_info.parquet
 43279811 2025-05-17 09:28:42.882232368 gene_orthologs.parquet
1019413239 2025-05-17 09:28:47.416233993 gene_refseq_uniprotkb_collab.parquet</code></pre>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/vjcitn/RNCBIGene/">Browse source code</a></li>
<li><a href="https://github.com/vjcitn/RNCBIGene/issues">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing RNCBIGene</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Vince Carey <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-4046-0063" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vince Carey.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
