<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>RNCBIGene: NCBI Gene annotation in parquet • RNCBIGene</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="RNCBIGene: NCBI Gene annotation in parquet">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">RNCBIGene</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/RNCBIGene.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/vjcitn/RNCBIGene/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="RNCBIGene_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="RNCBIGene_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="RNCBIGene_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="RNCBIGene_files/datatables-binding-0.33/datatables.js"></script><link href="RNCBIGene_files/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="RNCBIGene_files/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="RNCBIGene_files/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="RNCBIGene_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="RNCBIGene_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>RNCBIGene: NCBI Gene annotation in parquet</h1>
                        <h4 data-toc-skip class="author">Vincent J. Carey, stvjc at channing.harvard.edu</h4>
            
            <h4 data-toc-skip class="date">May 20, 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/RNCBIGene/blob/HEAD/vignettes/RNCBIGene.Rmd"><code>vignettes/RNCBIGene.Rmd</code></a></small>
      <div class="d-none name"><code>RNCBIGene.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Bioconductor’s annotation system has functioned for well over a decade as a pivotal resource for analysis and tool development. See the <a href="https://bioconductor.org/packages/AnnotationDbi" class="external-link">AnnotationDbi</a> documentation for details. AnnotationDbi and packages based on it use SQLite to manage organism-specific vocabularies and to resolve queries about mappings between identifier sets. For example, learn the Gene Ontology annotations for gene ORMDL3 in <em>Homo sapiens</em>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">go_orm</span> <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, key<span class="op">=</span><span class="st">"ORMDL3"</span>, keytype<span class="op">=</span><span class="st">"SYMBOL"</span>,</span>
<span>    columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO"</span>, <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">go_orm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   SYMBOL         GO EVIDENCE ONTOLOGY ENTREZID</span></span>
<span><span class="co">## 1 ORMDL3 GO:0002903      IMP       BP    94103</span></span>
<span><span class="co">## 2 ORMDL3 GO:0005515      IPI       MF    94103</span></span>
<span><span class="co">## 3 ORMDL3 GO:0005783      IDA       CC    94103</span></span>
<span><span class="co">## 4 ORMDL3 GO:0005789      IEA       CC    94103</span></span>
<span><span class="co">## 5 ORMDL3 GO:0005886      TAS       CC    94103</span></span>
<span><span class="co">## 6 ORMDL3 GO:0006672      IBA       BP    94103</span></span></code></pre>
<p>With RNCBIGene, all NCBI’s GO annotations for all organisms are in a single resource. The <code>geneFromCache</code> package will retrieve an Apache Parquet representation of the gene2go.gz text resource distributed at NCBI’s FTP site. For this query, we use the knowledge the ORMDL3 has Gene ID 94103.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn/RNCBIGene">RNCBIGene</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="va">go_ds</span> <span class="op">=</span> <span class="fu">arrow</span><span class="fu">::</span><span class="fu"><a href="https://arrow.apache.org/docs/r/reference/open_dataset.html" class="external-link">open_dataset</a></span><span class="op">(</span><span class="fu"><a href="../reference/geneFromCache.html">geneFromCache</a></span><span class="op">(</span><span class="st">"gene2go.parquet"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">lkor</span> <span class="op">=</span> <span class="va">go_ds</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">`#tax_id`</span><span class="op">==</span><span class="fl">9606</span>, <span class="va">GeneID</span><span class="op">==</span><span class="fl">94103</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/compute.html" class="external-link">collect</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">lkor</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21"],[9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606],[94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103,94103],["GO:0002903","GO:0005515","GO:0005783","GO:0005789","GO:0005886","GO:0006672","GO:0006672","GO:0006686","GO:0006940","GO:0010508","GO:0010508","GO:0017059","GO:0017059","GO:0030148","GO:0030667","GO:0035579","GO:0042552","GO:0061744","GO:0090156","GO:1900060","GO:1900182"],["IMP","IPI","IDA","IEA","TAS","IBA","IMP","IEA","IEA","IDA","IMP","IBA","IDA","IBA","TAS","TAS","IEA","IEA","IBA","IMP","IDA"],["acts_upstream_of_or_within","enables","located_in","located_in","located_in","involved_in","involved_in","involved_in","involved_in","acts_upstream_of_or_within","acts_upstream_of_or_within","part_of","part_of","involved_in","located_in","located_in","involved_in","involved_in","involved_in","acts_upstream_of_or_within","acts_upstream_of_or_within"],["negative regulation of B cell apoptotic process","protein binding","endoplasmic reticulum","endoplasmic reticulum membrane","plasma membrane","ceramide metabolic process","ceramide metabolic process","sphingomyelin biosynthetic process","regulation of smooth muscle contraction","positive regulation of autophagy","positive regulation of autophagy","serine palmitoyltransferase complex","serine palmitoyltransferase complex","sphingolipid biosynthetic process","secretory granule membrane","specific granule membrane","myelination","motor behavior","intracellular sphingolipid homeostasis","negative regulation of ceramide biosynthetic process","positive regulation of protein localization to nucleus"],["28747345","20182505|32296183|33558761","12093374","-","-","-","20182505","-","-","28747345","28747345","-","20182505|25691431","-","-","-","-","-","-","25691431","28747345"],["Process","Function","Component","Component","Component","Process","Process","Process","Process","Process","Process","Component","Component","Process","Component","Component","Process","Process","Process","Process","Process"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>#tax_id<\/th>\n      <th>GeneID<\/th>\n      <th>GO_ID<\/th>\n      <th>Evidence<\/th>\n      <th>Qualifier<\/th>\n      <th>GO_term<\/th>\n      <th>PubMed<\/th>\n      <th>Category<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"#tax_id","targets":1},{"name":"GeneID","targets":2},{"name":"GO_ID","targets":3},{"name":"Evidence","targets":4},{"name":"Qualifier","targets":5},{"name":"GO_term","targets":6},{"name":"PubMed","targets":7},{"name":"Category","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>It is not the purpose of this package to provide pin-compatible replacements for the AnnotationDbi packages. That would be a huge effort, and material support for such an effort does not exist at this time. This package is created to explore the opportunities for simplification and performance enhancement arising from the adoption of Parquet and Arrow for annotation representation and interrogation.</p>
</div>
<div class="section level2">
<h2 id="scope">Scope<a class="anchor" aria-label="anchor" href="#scope"></a>
</h2>
<p>There are seven parquet files that represent the NCBI Gene content available on February 22 2025.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">RNCBIGene</span><span class="fu">:::</span><span class="va">available_gene_parquet</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "gene2go.parquet"                     </span></span>
<span><span class="co">## [2] "gene2pubmed.parquet"                 </span></span>
<span><span class="co">## [3] "gene_info.parquet"                   </span></span>
<span><span class="co">## [4] "gene2accession.parquet"              </span></span>
<span><span class="co">## [5] "gene2refseq.parquet"                 </span></span>
<span><span class="co">## [6] "gene_orthologs.parquet"              </span></span>
<span><span class="co">## [7] "gene_refseq_uniprotkb_collab.parquet"</span></span></code></pre>
<p>Once we have cached them using <code>geneFromCache</code>, we can work with all of them “simultaneously”.</p>
<p>Because the retrieval process is laborious, the following chunk is not evaluated. Were it to be evaluated, it would be apparent that over 720 million records of annotation are available for interrogation.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">paths</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu">RNCBIGene</span><span class="fu">:::</span><span class="va">available_gene_parquet</span>, <span class="va">geneFromCache</span><span class="op">)</span></span>
<span><span class="va">pop</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="fu">arrow</span><span class="fu">::</span><span class="va"><a href="https://arrow.apache.org/docs/r/reference/open_dataset.html" class="external-link">open_dataset</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pop</span><span class="op">)</span> <span class="op">=</span> <span class="fu">RNCBIGene</span><span class="fu">:::</span><span class="va">available_gene_parquet</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">pop</span>, <span class="va">nrow</span><span class="op">)</span></span></code></pre></div>
<p>For illustration, the content corresponding to four human genes on chromosome 17 has been saved in the package.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">litp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"litparquet"</span>, package<span class="op">=</span><span class="st">"RNCBIGene"</span><span class="op">)</span>, full.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">litpop</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">litp</span>,</span>
<span>    <span class="fu">arrow</span><span class="fu">::</span><span class="va"><a href="https://arrow.apache.org/docs/r/reference/open_dataset.html" class="external-link">open_dataset</a></span><span class="op">)</span></span>
<span><span class="va">ns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">litp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">litpop</span><span class="op">)</span> <span class="op">=</span> <span class="va">ns</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">litpop</span>, <span class="va">nrow</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    lit.gene_info.parquet </span></span>
<span><span class="co">##                                        4 </span></span>
<span><span class="co">##               lit.gene_orthologs.parquet </span></span>
<span><span class="co">##                                     1546 </span></span>
<span><span class="co">## lit.gene_refseq_uniprotkb_collab.parquet </span></span>
<span><span class="co">##                                      986 </span></span>
<span><span class="co">##               lit.gene2accession.parquet </span></span>
<span><span class="co">##                                     3265 </span></span>
<span><span class="co">##                      lit.gene2go.parquet </span></span>
<span><span class="co">##                                      364 </span></span>
<span><span class="co">##                  lit.gene2pubmed.parquet </span></span>
<span><span class="co">##                                    15253 </span></span>
<span><span class="co">##                  lit.gene2refseq.parquet </span></span>
<span><span class="co">##                                      844</span></span></code></pre>
<p>Example: find the UniProt identifiers associated with gene ORMDL3 in humans.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pro</span> <span class="op">=</span> <span class="va">litpop</span><span class="op">[[</span><span class="st">"lit.gene2accession.parquet"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span> </span>
<span>   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">`#tax_id`</span> <span class="op">==</span> <span class="fl">9606</span>, <span class="va">Symbol</span> <span class="op">==</span> <span class="st">"ORMDL3"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">protein_accession.version</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/compute.html" class="external-link">collect</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="st">"-"</span><span class="op">)</span></span>
<span><span class="va">litpop</span><span class="op">[[</span><span class="st">"lit.gene_refseq_uniprotkb_collab.parquet"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">NCBI_tax_id</span> <span class="op">==</span> <span class="fl">9606</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">`#NCBI_protein_accession`</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">pro</span><span class="op">)</span>  <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/compute.html" class="external-link">collect</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 36 × 5</span></span></span>
<span><span class="co">##    `#NCBI_protein_accession` UniProtKB_protein_ac…¹ NCBI_tax_id UniProtKB_tax_id</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> NP_001307730.1            B3KS83                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> NP_001307730.1            J3QRM9                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> NP_001307730.1            Q6UY83                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> NP_001307730.1            Q8N138                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> NP_001307731.1            B3KS83                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> NP_001307731.1            J3QRM9                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> NP_001307731.1            Q6UY83                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> NP_001307731.1            Q8N138                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> NP_001307732.1            B3KS83                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> NP_001307732.1            J3QRM9                        <span style="text-decoration: underline;">9</span>606             <span style="text-decoration: underline;">9</span>606</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 26 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ abbreviated name: ¹​UniProtKB_protein_accession</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: method &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="applications">Applications<a class="anchor" aria-label="anchor" href="#applications"></a>
</h2>
<div class="section level3">
<h3 id="tidyverse-oriented-annotation-mapping">Tidyverse-oriented annotation mapping<a class="anchor" aria-label="anchor" href="#tidyverse-oriented-annotation-mapping"></a>
</h3>
<p>Typically one would use org.Hs.eg.db to remap identifiers from a SummarizedExperiment. We want to be able to do this for any identifier used in NCBI annotation, and for any organism. Let’s start with a dataset for human airway transcriptomics. The features are annotated with Ensembl gene identifiers, and we will add map locations and MIM codes using <code>mapIdsNG</code>, a variant of AnnotationDbi’s mapIds functionality for NCBI Gene.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">airway</span>, package<span class="op">=</span><span class="st">"airway"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"tidySummarizedExperiment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required namespace: tidySummarizedExperiment</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">airway</span>, <span class="st">"tidySummarizedExperiment"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">=</span> <span class="va">tse</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>map_location<span class="op">=</span><span class="fu"><a href="../reference/mapIdsNG.html">mapIdsNG</a></span><span class="op">(</span>keys<span class="op">=</span><span class="va">.feature</span>, keytype<span class="op">=</span><span class="st">"Ensembl"</span>, column<span class="op">=</span><span class="st">"map_location"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">=</span> <span class="va">tse</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>MIM<span class="op">=</span><span class="fu"><a href="../reference/mapIdsNG.html">mapIdsNG</a></span><span class="op">(</span>keys<span class="op">=</span><span class="va">.feature</span>, keytype<span class="op">=</span><span class="st">"Ensembl"</span>, column<span class="op">=</span><span class="st">"MIM"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">$</span><span class="va">map_location</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##               -         10p11.1        10p11.21  10p11.21-p11.1        10p11.22 </span></span>
<span><span class="co">##              12               8              29               1              24 </span></span>
<span><span class="co">## 10p11.22-p11.21 </span></span>
<span><span class="co">##               1</span></span></code></pre>
<p>Note that the identifiers that are conveniently available for <code>mapIdsNG</code> are currently limited to those in <code>gene_info</code>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">litpop</span><span class="op">[[</span><span class="st">"lit.gene_info.parquet"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "#tax_id"                              </span></span>
<span><span class="co">##  [2] "GeneID"                               </span></span>
<span><span class="co">##  [3] "Symbol"                               </span></span>
<span><span class="co">##  [4] "LocusTag"                             </span></span>
<span><span class="co">##  [5] "Synonyms"                             </span></span>
<span><span class="co">##  [6] "dbXrefs"                              </span></span>
<span><span class="co">##  [7] "chromosome"                           </span></span>
<span><span class="co">##  [8] "map_location"                         </span></span>
<span><span class="co">##  [9] "description"                          </span></span>
<span><span class="co">## [10] "type_of_gene"                         </span></span>
<span><span class="co">## [11] "Symbol_from_nomenclature_authority"   </span></span>
<span><span class="co">## [12] "Full_name_from_nomenclature_authority"</span></span>
<span><span class="co">## [13] "Nomenclature_status"                  </span></span>
<span><span class="co">## [14] "Other_designations"                   </span></span>
<span><span class="co">## [15] "Modification_date"                    </span></span>
<span><span class="co">## [16] "Feature_type"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">## Running under: macOS Sequoia 15.4.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] dplyr_1.1.4          RNCBIGene_0.0.4      org.Hs.eg.db_3.21.0 </span></span>
<span><span class="co">##  [4] AnnotationDbi_1.71.0 IRanges_2.43.0       S4Vectors_0.47.0    </span></span>
<span><span class="co">##  [7] Biobase_2.69.0       BiocGenerics_0.55.0  generics_0.1.4      </span></span>
<span><span class="co">## [10] BiocStyle_2.37.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] tidyselect_1.2.1                viridisLite_0.4.2              </span></span>
<span><span class="co">##  [3] farver_2.1.2                    blob_1.2.4                     </span></span>
<span><span class="co">##  [5] filelock_1.0.3                  arrow_20.0.0                   </span></span>
<span><span class="co">##  [7] Biostrings_2.77.1               lazyeval_0.2.2                 </span></span>
<span><span class="co">##  [9] fastmap_1.2.0                   BiocFileCache_2.99.4           </span></span>
<span><span class="co">## [11] duckdb_1.2.2                    digest_0.6.37                  </span></span>
<span><span class="co">## [13] lifecycle_1.0.4                 ellipsis_0.3.2                 </span></span>
<span><span class="co">## [15] tidySummarizedExperiment_1.19.0 KEGGREST_1.49.0                </span></span>
<span><span class="co">## [17] RSQLite_2.3.11                  magrittr_2.0.3                 </span></span>
<span><span class="co">## [19] compiler_4.5.0                  rlang_1.1.6                    </span></span>
<span><span class="co">## [21] sass_0.4.10                     tools_4.5.0                    </span></span>
<span><span class="co">## [23] utf8_1.2.5                      yaml_2.3.10                    </span></span>
<span><span class="co">## [25] data.table_1.17.2               knitr_1.50                     </span></span>
<span><span class="co">## [27] S4Arrays_1.9.0                  htmlwidgets_1.6.4              </span></span>
<span><span class="co">## [29] bit_4.6.0                       curl_6.2.2                     </span></span>
<span><span class="co">## [31] DelayedArray_0.35.1             RColorBrewer_1.1-3             </span></span>
<span><span class="co">## [33] abind_1.4-8                     withr_3.0.2                    </span></span>
<span><span class="co">## [35] purrr_1.0.4                     desc_1.4.3                     </span></span>
<span><span class="co">## [37] grid_4.5.0                      fansi_1.0.6                    </span></span>
<span><span class="co">## [39] ggplot2_3.5.2                   scales_1.4.0                   </span></span>
<span><span class="co">## [41] dichromat_2.0-0.1               SummarizedExperiment_1.39.0    </span></span>
<span><span class="co">## [43] cli_3.6.5                       rmarkdown_2.29                 </span></span>
<span><span class="co">## [45] crayon_1.5.3                    ragg_1.4.0                     </span></span>
<span><span class="co">## [47] httr_1.4.7                      DBI_1.2.3                      </span></span>
<span><span class="co">## [49] cachem_1.1.0                    stringr_1.5.1                  </span></span>
<span><span class="co">## [51] assertthat_0.2.1                BiocManager_1.30.25            </span></span>
<span><span class="co">## [53] XVector_0.49.0                  matrixStats_1.5.0              </span></span>
<span><span class="co">## [55] vctrs_0.6.5                     Matrix_1.7-3                   </span></span>
<span><span class="co">## [57] ttservice_0.4.1                 jsonlite_2.0.0                 </span></span>
<span><span class="co">## [59] bookdown_0.43                   bit64_4.6.0-1                  </span></span>
<span><span class="co">## [61] systemfonts_1.2.3               crosstalk_1.2.1                </span></span>
<span><span class="co">## [63] plotly_4.10.4                   tidyr_1.3.1                    </span></span>
<span><span class="co">## [65] jquerylib_0.1.4                 glue_1.8.0                     </span></span>
<span><span class="co">## [67] pkgdown_2.1.2                   DT_0.33                        </span></span>
<span><span class="co">## [69] stringi_1.8.7                   gtable_0.3.6                   </span></span>
<span><span class="co">## [71] GenomeInfoDb_1.45.3             GenomicRanges_1.61.0           </span></span>
<span><span class="co">## [73] UCSC.utils_1.5.0                tibble_3.2.1                   </span></span>
<span><span class="co">## [75] pillar_1.10.2                   rappdirs_0.3.3                 </span></span>
<span><span class="co">## [77] htmltools_0.5.8.1               R6_2.6.1                       </span></span>
<span><span class="co">## [79] dbplyr_2.5.0                    httr2_1.1.2                    </span></span>
<span><span class="co">## [81] textshaping_1.0.1               evaluate_1.0.3                 </span></span>
<span><span class="co">## [83] lattice_0.22-7                  png_0.1-8                      </span></span>
<span><span class="co">## [85] memoise_2.0.1                   bslib_0.9.0                    </span></span>
<span><span class="co">## [87] SparseArray_1.9.0               xfun_0.52                      </span></span>
<span><span class="co">## [89] fs_1.6.6                        MatrixGenerics_1.21.0          </span></span>
<span><span class="co">## [91] pkgconfig_2.0.3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vince Carey.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
